code/model/VirtualPage.php

```diff
@@ -441,22 +441,9 @@ public function __call($method, $args) {
 	 */
 	public function hasField($field) {
 		return (
-			array_key_exists($field, $this->record)
-			|| $this->hasDatabaseField($field)
-			|| array_key_exists($field, $this->db()) // Needed for composite fields
-			|| parent::hasMethod("get{$field}")
+			parent::hasField($field)
 			|| $this->CopyContentFrom()->hasField($field)
 		);
-	}	
```

Comment:
Yes I think it's needed. DataObject::hasMethod respects `get{$field}` so this needs to also.